x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -749,17 +757,36 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return MVT_SOURCE_LAYER_NAME;
   }
 
+  async _getEditableIndex(): Promise<string> {
+    const indexList = await this.getSourceIndexList();
+    if (indexList.length === 0) {
+      throw new Error(
+        i18n.translate('xpack.maps.source.esSearch.indexZeroLengthEditError', {
+          defaultMessage: `There are no indexes associated with this index pattern. Index pattern must be associated with a single index.`,
+        })
+      );
+    }
+    if (indexList.length > 1) {
+      throw new Error(
+        i18n.translate('xpack.maps.source.esSearch.indexOverOneLengthEditError', {
+          defaultMessage: `Multiple indexes are associated with this index pattern. Index pattern must be associated with a single index.`,
```

Comment:
Suggestion:

Your index pattern points to multiple indices. Only one index is allowed per index pattern.